### ISSUE TITLE
feature: visual tokenizer in webUI

### ISSUE THREAD

**Issue by blightbow** [jundot/omlx#562]
**Is your feature request related to a problem? Please describe.**
One of my favorite features from text-generation-webui was the ability to visually render the tokens based on the loaded model. While there are token calculators out there, this approach has the distinct advantage of running against the model that you already have loaded, and not being at the mercy of what the various token calculation libraries support. This is particularly helpful when a model leverages custom Python code in its tokenizer.

**Describe the solution you'd like**
Visually it would look something like this:
https://tokencounter.org

Tokencounter.org dynamically renders the tokens as the user types, but I'd be fine with the page being static until a button submits the text. I would definitely want it to support showing the numeric token ID mappings.

**Additional context**
I'm happy to submit a PR mockup for this feature, but I wanted to make sure this fit your vision for the project and understand where you would want this interface to live in the UI. Let me know!

**Comment by jundot** [comment]
The "chat with oMLX" ui was originally built as a basic interface for testing whether models load and respond correctly. There are already great open source chat clients out there like open webui or librechat for full-featured chat experiences.

I'm curious though, what kind of value would a visual tokenizer provide for the typical local ai beginner that omlx targets? For the chat interface improvements, i think features like web search or rag-based document search would serve a broader audience first.

Happy to discuss further if you have thoughts on how this would benefit everyday users!

**Comment by blightbow** [comment]
This was meant less for use with "chat with oMLX" and more as an administrative aid. You're correct that this would not be of much use to beginners. It's helpful when an experienced user wants to look up a token ID by number. It would be a useful companion feature to enabling `logit_bias` inside the API for banning tokens: https://help.openai.com/en/articles/5247780-using-logit-bias-to-alter-token-probability-with-the-openai-api

It dawns on me that this feature request was actually backwards; I hadn't checked to see if oMLX _supported_ `logit_bias` before opening the ticket, and just assumed that it already did. Consider this a two part request:

- Mind if I submit a PR for `logit_bias`?
- Alongside this, it would be helpful to have a way to look up token IDs without having to do so through code or submitting an API request.
  - It'd be nice if there was a form somewhere in oMLX that allows the user to type arbitrary text and perform token ID lookups against the loaded model. While it's not useful to beginners, it's the type of feature that power users would appreciate living in the GUI. (i.e. using a GUI for what it's good for)
  - If you're agreeable, the biggest question is where it would live. I don't want to build this into the wrong location of the UI and have to move it later.

**Comment by blightbow** [comment]
A quick Claude Code summary on how it's implemented in mlx_lm's reference [server.py](https://raw.githubusercontent.com/ml-explore/mlx-lm/refs/heads/main/mlx_lm/server.py):

> The server accepts an OpenAI-compatible `logit_bias` field in the request body — a `dict[int, float]` mapping token IDs to additive bias values (`server.py:1234`). Validation coerces keys to `int` and rejects anything that isn't a dict-of-int-to-float (`server.py:1308-1315`).
> 
> The value is carried through `LogitsProcessorArguments` (`server.py:382`) alongside the other sampling knobs (repetition/presence/frequency penalties, XTC), then handed to `make_logits_processors()` from `mlx_lm.sample_utils` (`server.py:612-617`).
> 
> The actual bias is applied as a **logits processor**, not at the sampling stage (`sample_utils.py:107-114`):
> 
> ```python
> if logit_bias:
>     indices = mx.array(list(logit_bias.keys()))
>     values = mx.array(list(logit_bias.values()))
> 
>     def logit_bias_processor(_, logits):
>         return logits.at[:, indices].add(values)
> 
>     logits_processors.append(logit_bias_processor)
> ```
> 
> So it's a scatter-add onto the logits tensor before sampling, composed into the same processor chain as the penalty processors. The processors are a `list[Callable[[tokens, logits], logits]]` plumbed through `stream_generate`/`BatchGenerator` — `BatchGenerator` also supports per-request processors via `insert(..., logits_processors=...)` (see `tests/test_generate.py:388-396`), which is the mechanism omlx would want for per-session bias in a batched server.
>
